doc/proposals/composable-integration-in sdk.md

```diff
@@ -0,0 +1,173 @@
+---
+title: Composable integration in SDK
+authors:
+  - "dettori@us.ibm.com"	
+  - "luan@us.ibm.com"
+  - "roytman@il.ibm.com"
+  - "tardieu@us.ibm.com"
+  - "mvaziri@us.ibm.com"
+  
+ 
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-12-16
+last-updated: 2019-12-16
+status: implemented
+see-also:
+replaces:
+superseded-by:
+---
+
+# Composable integration in SDK
+
+## Release Signoff Checklist
+
+- \[x\] Enhancement is `implementable`
+- \[x\] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created in [operator-sdk/doc][operator-sdk-doc]
+
+
+## Summary
+
+Kubernetes object specifications often require constant values for their fields. When deploying an entire application
+with many different resources, this limitation often results in the need for staged deployments, because some resources have to be deployed first in order to determine what data to provide for the specifications of dependent resources. This undermines the declarative nature of Kubernetes object specification and requires workflows, manual step-by-step instructions and/or brittle automated scripts for the deployment of applications as a whole.
```

Comment:
What does it mean for the referencing object and its lifecycle when the referenced object does not exist yet? Will it be created?

Reply:
@dmesser If the referenced object does not exist, then the referencing object's controller will keep retrying until that object is created. When that object is created, then the controller can resolve the referencing object and continue with whatever it needs to do.

Reply:
Interesting! How long would it keep trying?

Reply:
That depends on the consumer of the library. In the memcached tutorial, this is done indefinitely. 

We can imagine having a message in the Status of the object saying that it is waiting for some object to be created. If there is a bug in the spec of the object reference, the user could catch that and fix it. In any case, when the error is fixed or the object is created, then the controller can resolve the referencing object and move forward.